build/build-plugins.js

```diff
@@ -23,150 +25,105 @@ const plugins = [
-    case 'Tab':
-      return defaultPluginConfig
-
-    case 'Base':
-    case 'Button':
-    case 'Carousel':
-    case 'Collapse':
-    case 'Modal':
-    case 'ScrollSpy': {
-      const config = Object.assign(defaultPluginConfig)
-      config.external.push(bsPlugins.Manipulator)
-      config.globals[bsPlugins.Manipulator] = 'Manipulator'
-      return config
-    }
-
-    case 'Dropdown':
-    case 'Tooltip': {
-      const config = Object.assign(defaultPluginConfig)
-      config.external.push(bsPlugins.Manipulator, '@popperjs/core')
-      config.globals[bsPlugins.Manipulator] = 'Manipulator'
-      config.globals['@popperjs/core'] = 'Popper'
-      return config
-    }
+const domPlugins = {
+  Data: 'dom/data.js',
+  EventHandler: 'dom/event-handler.js',
+  Manipulator: 'dom/manipulator.js',
+  SelectorEngine: 'dom/selector-engine.js'
+}
 
-    case 'Popover':
-      return {
-        external: [
-          bsPlugins.Data,
-          bsPlugins.SelectorEngine,
-          bsPlugins.Tooltip
-        ],
-        globals: {
-          [bsPlugins.Data]: 'Data',
-          [bsPlugins.SelectorEngine]: 'SelectorEngine',
-          [bsPlugins.Tooltip]: 'Tooltip'
-        }
-      }
+const mainPlugins = {
+  BaseComponent: 'base-component.js',
+  Alert: 'alert.js',
+  Button: 'button.js',
+  Carousel: 'carousel.js',
+  Collapse: 'collapse.js',
+  Dropdown: 'dropdown.js',
+  Modal: 'modal.js',
+  Offcanvas: 'offcanvas.js',
+  Popover: 'popover.js',
+  ScrollSpy: 'scrollspy.js',
+  Tab: 'tab.js',
+  Toast: 'toast.js',
+  Tooltip: 'tooltip.js'
+}
 
-    case 'Toast':
-      return {
-        external: [
-          bsPlugins.Data,
-          bsPlugins.Base,
-          bsPlugins.EventHandler,
-          bsPlugins.Manipulator
-        ],
-        globals: {
-          [bsPlugins.Data]: 'Data',
-          [bsPlugins.Base]: 'Base',
-          [bsPlugins.EventHandler]: 'EventHandler',
-          [bsPlugins.Manipulator]: 'Manipulator'
-        }
-      }
+const nodePlugins = {
+  Popper: '@popperjs/core'
+}
 
-    default:
-      return {
-        external: []
-      }
-  }
+const bsPlugins = {
+  ...domPlugins,
+  ...utilPlugins,
+  ...mainPlugins
+}
+const allPlugins = {
+  ...bsPlugins,
+  ...nodePlugins
 }
 
-const utilObjects = new Set([
-  'Util',
-  'Sanitizer',
-  'Backdrop'
-])
+const makePluginPath = (pluginKey, basePath) => {
+  // eslint-disable-next-line no-prototype-builtins
+  if (nodePlugins.hasOwnProperty(pluginKey)) {
+    return nodePlugins[pluginKey]
+  }
 
-const domObjects = new Set([
-  'Data',
-  'EventHandler',
-  'Manipulator',
-  'SelectorEngine'
-])
+  const pluginFilename = allPlugins[pluginKey]
 
-const build = async plugin => {
-  console.log(`Building ${plugin} plugin...`)
+  return path.resolve(__dirname, `${basePath}/${pluginFilename}`)
+}
 
-  const { external, globals } = getConfigByPluginKey(plugin)
-  const pluginFilename = path.basename(bsPlugins[plugin])
-  let pluginPath = rootPath
+const resolvedDist = new Map()
```

Comment:
Nitpicking but shouldn't these 2 be `Set`s?

Reply:
I need both key and value, so preferred `Map`